packages/lesswrong/server/rsvpToEvent.ts

```diff
@@ -10,13 +10,27 @@ const responseSortOrder = {
   no: 3
 }
 
+type ValidResponse = 'yes' | 'no' | 'maybe';
+const isValidResponse = (response: string): response is ValidResponse => {
+  return response in responseSortOrder;
+}
+
 addGraphQLMutation('RSVPToEvent(postId: String, name: String, email: String, private: Boolean, response: String): Post');
 addGraphQLResolvers({
   Mutation: {
     async RSVPToEvent(root: void, {postId, name, email, nonPublic, response}: {postId: string, name: string, email: string, nonPublic: boolean, response: string}, context: ResolverContext) {
       const { currentUser } = context;
+      
+      if (!currentUser) {
```

Comment:
With the new generated types, Typescript was unhappy with the downstream `newRSVP` including `userId: currentUse?r._id` and being assigned to `rsvps`, since `userId` would be optional but `rsvps` expects a required `userId` field.

I just checked and we do allow rsvps from logged-out users.  `userId` is [marked as optional but not nullable](https://github.com/ForumMagnum/ForumMagnum/blob/master/packages/lesswrong/lib/collections/posts/schema.tsx#L61), which, in that context, seems wrong?